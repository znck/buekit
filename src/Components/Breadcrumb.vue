<script>
import PropTypes from '@znck/prop-types'
import { style, styleResolver, isIgnoredVnode } from '../utils'
import { createTag } from '../mixins'
import { sizes } from '../shared'


export default {
  name: 'Breadcrumb',
  functional: true,
  mixins: [createTag('nav')],
  props: {
    centered: PropTypes.bool.value(false),
    pushed: PropTypes.bool.value(false),
    separator: PropTypes.oneOf('succeeds', 'dot', 'bullet', 'arrow'),
    size: PropTypes.oneOf(sizes)
  },
  render (h, ctx) {
    const { size, tag, centered, pushed, separator } = ctx.props
    const _ = styleResolver(ctx.$style)
    const slots = ctx.slots()
    const links = (slots.default || []).filter(v => !isIgnoredVnode(v))
    const k = links.length - 1
    const wrapper = [
      h('ul', {}, links.map((link, i) => h('li', { class: [ i === k && _('is-active') ] }, [link])))
    ]

    return style(
      h(tag, ctx.data, wrapper),
      _('breadcrumb'),
      size && _(`is-${size}`),
      centered && _('is-centered'),
      pushed && _('is-right'),
      separator && _(`has-${separator}-separator`)
    )
  }
}
</script>

<style lang="scss" module>
@import '~bulma/sass/utilities/_all.sass';
@import '../_base.scss';
@import '~bulma/sass/components/breadcrumb.sass';
</style>
