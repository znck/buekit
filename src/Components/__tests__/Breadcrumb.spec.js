import {shallow} from 'vue-test-utils'
import {createRenderer} from 'vue-server-renderer'

import Breadcrumb from '../Breadcrumb.vue'

describe('Breadcrumb.vue', () => {
	it('matches snapshot', () => {
		const wrapper = shallow(Breadcrumb, { context: { props: {} } })
		const renderer = createRenderer()

		renderer.renderToString(wrapper.vm, (err, str) => {
			if (err) {
				throw new Error(err)
			}
			expect(str).toMatchSnapshot()
		})
	})
})
