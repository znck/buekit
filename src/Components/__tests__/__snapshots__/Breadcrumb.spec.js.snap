// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb.vue matches snapshot 1`] = `
<nav class="breadcrumb">
    <ul></ul>
</nav>
`;
